### PR TITLE
add version defaulting for the container runtime

### LINKED ISSUE #35: Make container runtime version optional
Based upon the entered Kubernetes version and the selected OS we should default to a docker/cri-o version.

For now the logic should be:
- cri-o
  - Kubernetes v1.8 + Ubuntu 16.04 -> error as theres no cri-o 1.8 package in the repos
  - Kubernetes v1.9 + Ubuntu 16.04 -> cri-o 1.9
  - Kubernetes v1.8 + Container Linux -> error as theres no cri-o for coreos
  - Kubernetes v1.8 + Container Linux -> error as theres no cri-o for coreos
- docker
  - Kubernetes v1.8 + Ubuntu 16.04 -> docker 1.13
  - Kubernetes v1.9 + Ubuntu 16.04 -> docker 1.13
  - Kubernetes v1.8 + Container Linux -> docker 1.12
  - Kubernetes v1.8 + Container Linux -> docker 1.12

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Adds sane defaulting for the container runtime.
Also it handles versions defined with and without `v` as prefix.

fixes #35

cc @alvaroaleman

### CODE REVIEW

File: docs/container_runtime.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Container Runtimes
+
+## Defaulting
+The machine-controller is able to default to a supported container runtime in case no runtime was specified in the machine-spec.
+Also in case only the version is missing only that will get defaulted.
```
Comment (metalmatze) [inline docs/container_runtime.md]: What is this sentence supposed to say?

File: pkg/containerruntime/crio/supported_versions.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+// GetOfficiallySupportedVersions returns the officially supported cri-o version for the given kubernetes version
+func GetOfficiallySupportedVersions(kubernetesVersion string) ([]string, error) {
+	v, err := semver.NewVersion(kubernetesVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	majorMinorString := fmt.Sprintf("%d.%d", v.Major(), v.Minor())
+	switch majorMinorString {
```
Comment (metalmatze) [inline pkg/containerruntime/crio/supported_versions.go]: Wouldn't be a `strings.HasPrefix(kubernetesVersion, "1.8")` be sufficient? But I don't have a problem with using this lib.

File: docs/container_runtime.md (hunk 1 above)
Comment (mrIncompetent) [reply docs/container_runtime.md]: You can simply define `machine.Spec.ContainerRuntime.Name` without defining `machine.Spec.ContainerRuntime.Version`

Comment (mrIncompetent) [reply docs/container_runtime.md]: I'll clarify that sentence

File: pkg/containerruntime/crio/supported_versions.go (hunk 2 above)
Comment (mrIncompetent) [reply pkg/containerruntime/crio/supported_versions.go]: It prevents breaking the code when someone enters `v1.9.2` or `1.9.2`

File: pkg/errors/errors.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package errors
+
+import "errors"
+
+var (
+	NoSupportedVersionAvailableErr = errors.New("no supported version available")
+	VersionNotAvailableErr         = errors.New("version not available")
+)
```
Comment (metalmatze) [inline pkg/errors/errors.go]: Can't we find a more well suited place for this file to be placed in? I'm hesitant with such files, as they just introduce more little helper packages. Either this error is returned by a package and also in that package, or it is consumed by a package and inside that package. 

For example `VersionNotAvailableErr` is only used in the `package ubuntu` so place it there.

File: pkg/containerruntime/crio/supported_versions.go (hunk 2 above)
Comment (metalmatze) [reply pkg/containerruntime/crio/supported_versions.go]: Fair enough. Then using that lib is probably better then starting your own regex magic. :sparkles:

File: pkg/errors/errors.go (hunk 3 above)
Comment (mrIncompetent) [reply pkg/errors/errors.go]: fixed in 3e8b884

**Pull request merged**